apps/web/src/app/(signing)/sign/[token]/signature-field.tsx

```diff
@@ -61,9 +61,16 @@ export const SignatureField = ({ field, recipient }: SignatureFieldProps) => {
     return 'signed-text';
   }, [field.inserted, signature?.signatureImageAsBase64]);
 
+  useEffect(() => {
+    if (showSignatureModal == false && isDocumentSigned == false) {
+      setLocalSignature(null);
+      return;
+    }
+  }, [showSignatureModal]);
```

Comment:
I think you should include all the required items in the dependency array

Other points:
- return is redundant
- `===` is preferred over `==` 
- `!value` is also generally preferred

```
  useEffect(() => {
    if (!showSignatureModal && !isDocumentSigned) {
      setLocalSignature(null);
    }
  }, [showSignatureModal, isDocumentSigned]);
```

Could you also add some spacing below this useEffect?